typing/typecore.ml

```diff
@@ -3000,11 +3015,17 @@ let rec type_approx env sexp =
       ({ pexp_desc = Pexp_extension({txt = "extension.escape"}, PStr []) },
        [Nolabel, e]) ->
     type_approx env e
-  | _ -> newvar ()
+  | _ -> newvar Type_layout.any
 
 (* Check that all univars are safe in a type. Both exp.exp_type and
    ty_expected should already be generalized. *)
 let check_univars env kind exp ty_expected vars =
+  let error ty ty_expected errs =
+    let ty_expected = instance ty_expected in
```

Comment:
This instance is just the one that was previously on line 3027.  I agree it's not obvious what we need it for, but error messages do change if we remove it, and I didn't think it was worth thinking through whether I liked the old ones or the new ones better, since it's orthogonal to the present PR.